packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2896 @@
+
+  /// The text style of the button that opens the submenu.
+  ///
+  /// The color in this text style will only be used if [buttonOverlayColor]
+  /// is unset.
+  final MaterialStateProperty<TextStyle?>? buttonTextStyle;
+
+  /// Called when the button that opens the submenu is hovered over.
+  final ValueChanged<bool>? onHover;
+
+  /// A callback that is invoked when the menu is opened.
+  final VoidCallback? onOpen;
+
+  /// A callback that is invoked when the menu is closed.
+  final VoidCallback? onClose;
+
+  /// The list of widgets that appear in the menu when it is opened.
+  ///
+  /// These can be any widget, but are typically either [MenuItemButton] or
+  /// [MenuButton] widgets.
+  final List<Widget> children;
+
+  @override
+  State<MenuButton> createState() => _MenuButtonState();
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    return <DiagnosticsNode>[
+      ...children.map<DiagnosticsNode>((Widget child) {
+        return child.toDiagnosticsNode();
+      })
+    ];
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<String>('label', label.toString(), defaultValue: null));
+    properties.add(DiagnosticsProperty<Widget>('leadingIcon', leadingIcon, defaultValue: null));
+    properties.add(DiagnosticsProperty<Widget>('trailingIcon', trailingIcon, defaultValue: null));
+    properties.add(StringProperty('semanticLabel', semanticsLabel, defaultValue: null));
+    properties.add(
+        DiagnosticsProperty<MaterialStateProperty<Color?>>('backgroundColor', backgroundColor, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<ShapeBorder?>>('shape', shape, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<double?>>('elevation', elevation, defaultValue: null));
+    properties.add(DiagnosticsProperty<EdgeInsetsDirectional?>('padding', padding, defaultValue: null));
+    properties.add(DiagnosticsProperty<EdgeInsetsDirectional?>('buttonPadding', buttonPadding, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<Color?>>('buttonBackgroundColor', buttonBackgroundColor,
+        defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<Color?>>('buttonForegroundColor', buttonForegroundColor,
+        defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<Color?>>('buttonOverlayColor', buttonOverlayColor,
+        defaultValue: null));
+    properties
+        .add(DiagnosticsProperty<MaterialStateProperty<ShapeBorder?>>('buttonShape', buttonShape, defaultValue: null));
+    properties.add(
+        DiagnosticsProperty<MaterialStateProperty<TextStyle?>>('buttonTextStyle', buttonTextStyle, defaultValue: null));
+  }
+}
+
+class _MenuButtonState extends State<MenuButton> {
+  late _ChildMenuNode entry;
+  MenuEntry? childMenu;
```

Comment:
Here and elsewhere: When reading this I found it confusing that `entry` is not a MenuEntry, but a MenuNode. Maybe unify terminology?